packages/dashboard/src/pages/settings.page.tsx

```diff
@@ -489,49 +521,101 @@ function SendGridConfig() {
     },
   });
 
+  const { showPassword, secretValue, setSecretValue, apiHandler, handleClickShowPassword, handleMouseDownPassword, upsertSecretRequest } = secretsEditor
   const requestInProgress =
-    sendgridProviderRequest.type === CompletionStatus.InProgress;
+    sendgridProviderRequest.type === CompletionStatus.InProgress || upsertSecretRequest.type === CompletionStatus.InProgress;
+
+  const onSubmit = () => {
+    apiHandler()
+    submitApiKey()
+  }
```

Comment:
Grouped these to tidy up the UI. 
My intention is to check if the values (secretValue & apiKey) changed before conditionally calling their api handler

```javascript
if (apiKeyChanged) submitApiKey()
else if (secretValueChanged) submitSecretValue()
```